### ISSUE TITLE
Create namespaces that collect all the appropriate driver methods into one spot

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#14486]
Writing a driver like `:vertica` for example requir

**Comment by dpsutton** [comment]
i think it would be nicer to not use potemkin here. I think if there was a driver.api namespace with clear docstrings and clear methods to be implemented it would be far easier for people to create drivers. Its true that `(dir metabase.driver)` would show this but i think its far more likely that people will read the namespace and having them defined there rather than brought in by potemkin sounds better.

**Comment by camsaul** [comment]
@dpsutton Fair point. We're already doing that for non-SQL drivers -- all of the multimethods for non-SQL drivers is in `metabase.driver`.  For SQL and SQL-JDBC drivers it's a different story. QP-related methods like `->honeysql` are in `metabase.driver.sql.query-processor`; JDBC sync methods are in `metabase.driver.sql-jdbc.sync`, and so forth. They're kind of all over the place. 

We could consolidate  all the SQL and SQL-JDBC methods into `metabase.driver.sql.interface` and `metabase.driver.sql-jdbc.interface` namespaces respectively. Maybe using Potemkin in reverse so those interface methods are still available in the old namespaces would make sense so we could give everybody some time to transfer over
